addon/utils/is-fastboot.js

```diff
@@ -0,0 +1,14 @@
+import { computed } from '@ember/object';
+import { getOwner } from '@ember/application';
+import { assert } from '@ember/debug';
+
+export default function isFastBoot() {
+  return computed(function() {
+    const container = getOwner(this);
+    assert('You may only use isFastBoot() on a container-aware object', container && container.lookup === 'function');
+    const fastboot = container.lookup('service:fastboot');
+    assert('You may only use isFastBoot() when with a fastboot service present');
```

Comment:
I think @marcoow was talking about returning false rather than throwing an assertion error

Reply:
Agree, although the cases where that may happen are quite limited now that he’s pulling the service directly from the container. 

Still — could be a useful error if you’re working in an engine whose container doesn’t have the service.